utils/tagSelectedNode.ts

```diff
@@ -0,0 +1,14 @@
+import { Node } from "reactflow";
+
+/**
+ * Make sure selected node has `selected` set to `true`
+ */
+export const tagSelectedNode = <T, U>(
+  nodes: Node<T>[],
+  selectedNode: Node<U> | undefined
+) => {
```

Comment:
The Node type parameter should be NodeDataCommon as those are the interactable nodes in the flow

Reply:
Yes, but in this function we don't care what the types of `T` and `U` are. The idea is to make `tagSelectedNode` as generic as possible to avoid involving a larger context than necessary. It also makes it easier to rewrite the rest of the application in the future, as this function is not tightly coupled to the rest of the business logic